From: Takashi Iwai <tiwai@suse.de>
Subject: [PATCH] Documentation/sphinx: Fix Directive import error
Date: Fri, 02 Mar 2018 12:49:03 +0100
Message-id: <s5hbmg6hf4g.wl-tiwai@suse.de>
Patch-mainline: Submitted, linux-doc ML
References: bsc#1083694

The sphinx.util.compat Directive stuff was deprecated in the recent
Sphinx version, and now we get a build error.

Let's take a fallback to the newer one, from docutils.parsers.rst.

Bugzilla: https://bugzilla.opensuse.org/show_bug.cgi?id=1083694
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
---
 Documentation/sphinx/kerneldoc.py |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/Documentation/sphinx/kerneldoc.py
+++ b/Documentation/sphinx/kerneldoc.py
@@ -37,7 +37,10 @@ import glob
 from docutils import nodes, statemachine
 from docutils.statemachine import ViewList
 from docutils.parsers.rst import directives
-from sphinx.util.compat import Directive
+try:
+    from sphinx.util.compat import Directive
+except ImportError:
+    from docutils.parsers.rst import directives, Directive
 from sphinx.ext.autodoc import AutodocReporter
 
 __version__  = '1.0'
